app/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -141,7 +139,7 @@ internal constructor(
   private val taskViewModels: MutableStateFlow<MutableMap<String, AbstractTaskViewModel>> =
     MutableStateFlow(mutableMapOf())
 
-  private val data: MutableMap<Task, TaskData?> = LinkedHashMap()
+  private val taskDataHandler = TaskDataHandler()
```

Comment:
is it a good place for object creation?

Reply:
We later need this as a dependency for TaskSequenceHandler which will also be initialized at class level.